### PR TITLE
Pr/v1.11 backport 2023 01 24

### PR DESCRIPTION
**Pull request by ldelossa**
- [ ] #22370 -- Bugtool: add flag to exclude object for endpoints (@tbalthazar)
- [x] #22726 -- test/helpers: Fix retry condition for CiliumExecContext (@christarazi)
- [x] #22654 -- backporting: leave `backport/author` PRs alone (@bimmlerd)
- [ ] #22619 -- fqdn: dnsproxy: fix data race in dns proxy implementation (@aspsk)
- [x] #22772 -- test: service: fix formatting of error msg in doFragmentedRequest() (@julianwiedmann)
- [ ] #22821 -- Add sphinxcontrib-googleanalytics to doc requirements (@chalin)
- [ ] #22600 -- daemon/cmd: improve stale cilium endpoint error handling. (@tommyp1ckles)
- [ ] #22807 -- Update Layer 7 Protocol Visibility Document. (@obaranov1)
- [ ] #22884 -- Fix crash of CES queue delay metric when CESTracker is nil (@dlapcevic)
- [ ] #22962 -- clustermesh: Add missing brackets of etcd option (@YutaroHayakawa)
- [x] #23095 -- docs: Improve wording for deny policies limitation (@joestringer)
- [ ] #23135 -- docs: Improve IPsec guide (@pchaigno)
- [ ] #23140 -- build(deps): update package dependencies (@fengshunli)
- [ ] #23134 -- docs: update committer security requirements (@xmulligan)
- [x] #23058 -- contrib: Update PR template for backport (@sayboras)
- [ ] #23158 -- IPsec: Refactor `ipSecReplaceState{In,Out}` functions (@pchaigno)
- [ ] #23267 -- Update CNI to 1.2.0 (@michi-covalent)
- [x] #23220 -- .github: Pin docker buildx version to v0.9.1 (v2) (@joestringer)
    - Had to drop the first commit.

Drop PRs:
 * #22620 -- Update Cilium install guide about EKS aws-node DaemonSet potential connectivity problem on uninstall (@NikAleksandrov)
    - ciliumendpoints.go does not exist in v1.11

 * #22220 -- Enable Google Analytics 4 (@chalin)
    - see: https://github.com/cilium/cilium/pull/23301#issuecomment-1402342887
  
  * #20619 -- test/k8s: remove l7_demos test (@tklauser)
    - appears irrelevant for back porting, workfile files do not exist.

Once this PR is merged, you can update the PR labels via:
```upstream-prs
$ for pr in 22370 22726 22654 22619 22772 22821 22600 22807 22884 22962 23095 23135 23140 23134 23058 23158 23220 23267 23220; do contrib/backporting/set-labels.py $pr done 1.11; done
```

### CODE REVIEW

**Review by sayboras — APPROVED** [review]
LGTM on my commits, thanks a lot :100:

**Comment by ldelossa** [comment]
Manually added image bump: 0c9f929619c9f00fd8c2df2d5510eb75311e7bbb

**Review by joestringer — APPROVED** [review]
For my commits, github-sec team

File: CODEOWNERS
Comment (joestringer) [inline CODEOWNERS]: None of these codeowners changes should be backported.

**Review by joestringer — CHANGES_REQUESTED** [review]

File: CODEOWNERS
Comment (joestringer) [reply CODEOWNERS]: For older branches, we only set codeowners for APIs, security and otherwise tophat. Where did these get bumped?

Comment (ldelossa) [reply CODEOWNERS]: Commit dropped.

**Review by joestringer — APPROVED** [review]

File: CODEOWNERS
Comment (michi-covalent) [reply CODEOWNERS]: dropped https://github.com/cilium/cilium/pull/20619 ✅ these files don't exist in v1.11 branch.

**Comment by joestringer** [comment]
Failure looks related to the backport commits:

https://github.com/cilium/cilium/actions/runs/4001219389/jobs/6867238521

```
# Generate the corresponding mock library.
docker container run --rm --workdir /src/mock --volume /home/runner/work/cilium/cilium/bpf/mock/..:/src --user "1000:1000" cilium/bpf-mock ./mock_customized.sh conntrack_stub.h;
Creating mock for conntrack_stub...
make[2]: Leaving directory '/home/runner/work/cilium/cilium/bpf/mock'
docker container run --rm --workdir /src/test/bpf --volume /home/runner/work/cilium/cilium:/src --user "1000:1000" cilium/bpf-mock  clang -I../../bpf/ -I../../bpf/include -I. -D__NR_CPUS__=4 -O2 -I../../bpf/mock -I ./../bpf/ -I../../../CMock/src -I../../../CMock/vendor/unity/src -I../../../hashmap/include  nat-test.c ../../../CMock/vendor/unity/src/unity.c ../../../CMock/src/cmock.c ../../../hashmap/src/hashmap.c ../../bpf/mock/mocks/mock_helpers.c ../../bpf/mock/mocks/mock_helpers_skb.c ../../bpf/mock/mocks/mock_conntrack_stub.c -o nat-test
clang: error: no such file or directory: '../../bpf/mock/mocks/mock_helpers_skb.c'
make[1]: *** [Makefile:49: nat-test] Error 1
make[1]: Leaving directory '/home/runner/work/cilium/cilium/test/bpf'
make: *** [Makefile:23: build] Error 2
make: Leaving directory '/home/runner/work/cilium/cilium/test'
warning: execution failed: make -C test build -j 4
You can fix the problem, and then run

  git rebase --continue
```

**Comment by michi-covalent** [comment]
struggle continues

**Review by YutaroHayakawa — APPROVED** [review]

**Review by julianwiedmann — APPROVED** [review]

**Review by tommyp1ckles — APPROVED** [review]

**Review by christarazi — APPROVED** [review]
Same comment as https://github.com/cilium/cilium/pull/23310#discussion_r1085933742.

File: CODEOWNERS
Comment (bimmlerd) [reply CODEOWNERS]: I think dropping all of #20619 is a bit much- yes the codeowners change can be dropped but https://github.com/cilium/cilium/blob/v1.11/test/k8sT/l7_demos.go does exist (note `test/k8sT` vs `test/k8s`), and so do the manifests.

**Review by bimmlerd — APPROVED** [review]
https://github.com/cilium/cilium/pull/22654 LGTM

**Comment by aanm** [comment]
/test-backport-1.11

**Comment by ldelossa** [comment]
@tklauser probably easiest to send another PR with backport.

**Comment by tklauser** [comment]
> @tklauser probably easiest to send another PR with backport.

https://github.com/cilium/cilium/pull/23348

**Comment by aanm** [comment]
**The entire CI was green so we can merge this PR once the reviews are in.**

**Review by michi-covalent — APPROVED** [review]

**Review by nbusseneau — APPROVED** [review]

File: CODEOWNERS
Comment (joestringer) [reply CODEOWNERS]: :+1: https://github.com/cilium/cilium/pull/23313#issuecomment-1403713116

**Pull request merged**
